File: services/geoapi/geoapi/Models/AgentInfo.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Newtonsoft.Json;

namespace geoapi.Models
{
    public class AgentInfo
    {
        [JsonIgnore]
        public ObjectId Id { get; set; }

        public string AgentToken { get; set; }

        public string AgentName { get; set; }

        [JsonIgnore]
        [BsonElement("expireAt")]
        public DateTime ExpireAt;

        public AgentInfo(string agentToken, string agentName, DateTime expireAt)
        {
            AgentToken = agentToken;
            AgentName = agentName;
            ExpireAt = expireAt;
        }
    }
}

File: services/geoapi/geoapi/Models/AgentInfo.vb
Imports System
Imports MongoDB.Bson
Imports MongoDB.Bson.Serialization.Attributes
Imports Newtonsoft.Json

Namespace Models
    Public Class AgentInfo
        <JsonIgnore>
        Public Property Id As ObjectId
        Public Property AgentToken As String
        Public Property AgentName As String
        <JsonIgnore>
        <BsonElement("expireAt")>
        Public ExpireAt As DateTime

        Public Sub New(ByVal agentToken As String, ByVal agentName As String, ByVal expireAt As DateTime)
            Me.AgentToken = agentToken
            Me.AgentName = agentName
            Me.ExpireAt = expireAt
        End Sub
    End Class
End Namespace
